scripts/cp-pkgs.js

```diff
@@ -74,12 +90,67 @@ function cpAsset(asset) {
     Promise.reject(new Error(`Non-existent asset package path ${assetPkg} for ${asset}`));
   }
   const destDir = path.join(assetPkg, 'dist', path.basename(asset));
-  return cpFile(asset, destDir).then(() => console.log(`cp ${asset} -> ${destDir}`));
+  return cpFile(asset, destDir).then(
+    () => console.log(`cp ${asset} -> ${destDir}`),
+    (err) => {
+      throw err;
+    }
+  );
 }
 
-cleanPkgDistDirs();
+function dtsBundler() {
+  const packageDirectories = fs.readdirSync(D_TS_DIRECTORY);
+  packageDirectories.forEach((packageDirectory) => {
+    const main = path.resolve(D_TS_DIRECTORY, packageDirectory, './index.d.ts');
+    dts.bundle({
+      name: packageDirectory,
+      main,
+    });
+  });
+}
 
-Promise.all(globSync('build/*.{css,js,map}').map(cpAsset)).catch((err) => {
-  console.error(`Error encountered copying assets: ${err}`);
-  process.exit(1);
-});
+function getDeclarationFileName(asset, assetPkg) {
+  const packageName = path.parse(asset).name.replace(/^mdc-|\.d$/g, '');
+  const isAllInOne = packageName === ALL_IN_ONE_PACKAGE;
+  const destFileName = isAllInOne ? packageName : `mdc.${toCamelCase(packageName)}`;
+  return path.join(assetPkg, 'dist', `${destFileName}.d.ts`);
+}
+
+function cpDeclarationAsset(asset) {
```

Comment:
Mark this function `async` and add a JSDoc annotation so we know it returns a `Promise<whatever>`